context.go

```diff
@@ -72,6 +73,22 @@ func GetHeader(ctx Context) (abci.Header, bool) {
 	return val, ok
 }
 
+// WithCommitInfo sets the info on who signed the block in this Context.
+// Panics if already set.
+func WithCommitInfo(ctx Context, info abci.LastCommitInfo) Context {
```

Comment:
I try not to use any abci/tendermint data structures in the weave internals to provide a loose coupling, also so all structs are documented in our codebase (I need to check this out, and follow imports into tendermint to even see what is in abci.LastCommitInfo).

Can you just make an abci.LastCommitInfo -> weave.CommitInfo helper as in genesis file, and then we can discuss if all fields are needed? Or anything else?

Reply:
Sounds good to me, I wanted to validate the approach with you first.
I'm kind of ambivalent about the approach that should be taken, because our app is still pretty tightly coupled to tendermint in terms of interfaces/types it should support. One good reason to use tendermint structs is that when they change any of their types with tendermint upgrade - we get some if not all of that update for free.

In terms of digging to deps in order to see what they do, I guess I'm too spoiled using a fully fledged IDE to even notice, but it might definitely help a lot with IDEless PR reviews as well as means of documenting stuff. The downside is that we have to follow tendermints objects and duplicate the impl.